src/m365/spo/commands/homesite/homesite-set.ts

```diff
@@ -24,42 +44,15 @@ class SpoHomeSiteSetCommand extends SpoCommand {
     return 'Sets the specified site as the Home Site';
   }
 
-  constructor() {
-    super();
-    this.#initTelemetry();
-    this.#initOptions();
-    this.#initValidators();
-    this.#initTypes();
+  public get schema(): z.ZodTypeAny {
+    return optionsSchema;
   }
 
-  #initTelemetry(): void {
-    this.telemetry.push((args: CommandArgs) => {
-      Object.assign(this.telemetryProperties, {
-        vivaConnectionsDefaultStart: typeof args.options.vivaConnectionsDefaultStart !== 'undefined'
+  public getRefinedSchema(schema: z.ZodTypeAny): z.ZodEffects<any> | undefined {
+    return schema
+      .refine((options: Options) => !options.audienceIds || [options.audienceIds, options.audienceNames].filter(o => o !== undefined).length === 1, {
+        message: 'Use one of the following options when specifying the audience name: audienceIds or audienceNames.'
       });
-    });
-  }
-
-  #initOptions(): void {
-    this.options.unshift(
-      {
-        option: '-u, --siteUrl <siteUrl>'
-      },
-      {
-        option: '--vivaConnectionsDefaultStart [vivaConnectionsDefaultStart]',
-        autocomplete: ['true', 'false']
-      }
-    );
-  }
-
-  #initValidators(): void {
-    this.validators.push(
-      async (args: CommandArgs) => validation.isValidSharePointUrl(args.options.siteUrl)
-    );
-  }
-
-  #initTypes(): void {
-    this.types.boolean.push('vivaConnectionsDefaultStart');
   }
 
   public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
```

Comment:
Let's also update the first verbose message used below. This command will not set the home site anymore, only update existing ones.